### PR TITLE
remove deprecated ServiceDiscovery.Instances

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
1. stop using `Instances`, and use `InstancesByPort`

1. remove this deprecated method

### CODE REVIEW

File: pilot/pkg/serviceregistry/aggregate/controller.go [hunk 1]
```diff
@@ -131,7 +131,7 @@ func (c *Controller) Services() ([]*model.Service, error) {
 					// First time we see a service. The result will have a single service per hostname
 					// The first cluster will be listed first, so the services in the primary cluster
 					// will be used for default settings. If a service appears in multiple clusters,
-					// the order is less clear.
+					// the order is less clear.clusterAddressesMutex
```
Comment (ZackButcher) [inline pilot/pkg/serviceregistry/aggregate/controller.go]: copy/paste error?

Comment (hzxuzhonghu) [reply pilot/pkg/serviceregistry/aggregate/controller.go]: oh, yes, will remove it.

**Comment by hzxuzhonghu** [comment]
@ZackButcher addressed

**Pull request merged**